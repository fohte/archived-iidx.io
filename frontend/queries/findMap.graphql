query findMap($id: ID!, $playStyle: PlayStyle!, $difficulty: Difficulty!) {
  music(id: $id) {
    id
    title
    subTitle
    genre
    artist
    textageUid
    series
    leggendaria
    map(playStyle: $playStyle, difficulty: $difficulty) {
      id
      numNotes
      level
      playStyle
      difficulty
      minBpm
      maxBpm
    }
  }
}
