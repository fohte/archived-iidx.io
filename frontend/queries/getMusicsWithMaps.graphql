query getMusicsWithMaps {
  musics {
    __typename
    id
    title
    subTitle
    maps {
      __typename
      id
      difficulty
      level
      playStyle
    }
  }
}
