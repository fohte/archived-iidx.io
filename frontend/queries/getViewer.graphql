query getViewer {
  viewer {
    __typename
    id
    uid
  }
}
