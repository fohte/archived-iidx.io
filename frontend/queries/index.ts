/* tslint:disable */

export interface Query {
  musics?: Music[] | null /** Find musics. */
  user?: User | null /** Find a user by ID. */
  viewer?: User | null /** The currently authenticated user. */
}

export interface Music {
  artist: string
  genre: string
  id: string
  leggendaria: boolean
  maps: Map[]
  series: number
  subTitle: string
  textageUid: string
  title: string
}

export interface Map {
  difficulty: Difficulty
  id: string
  level: number
  maxBpm: number
  minBpm: number
  numNotes: number
  playStyle: PlayStyle
}

export interface User {
  id: string
  profile: UserProfile
  uid: string
}

export interface UserProfile {
  name: string
}

export interface Mutation {}
export interface UserQueryArgs {
  id: string
}

export enum Difficulty {
  NORMAL = 'NORMAL',
  HYPER = 'HYPER',
  ANOTHER = 'ANOTHER',
}

export enum PlayStyle {
  SP = 'SP',
  DP = 'DP',
}

export type GetMusicsWithMapsVariables = {}

export type GetMusicsWithMapsQuery = {
  __typename?: 'Query'
  musics?: GetMusicsWithMapsMusics[] | null
}

export type GetMusicsWithMapsMusics = {
  __typename?: 'Music'
  id: string
  title: string
  subTitle: string
  maps: GetMusicsWithMapsMaps[]
}

export type GetMusicsWithMapsMaps = {
  __typename?: 'Map'
  id: string
  difficulty: Difficulty
  level: number
  playStyle: PlayStyle
}

export type GetViewerVariables = {}

export type GetViewerQuery = {
  __typename?: 'Query'
  viewer?: GetViewerViewer | null
}

export type GetViewerViewer = {
  __typename?: 'User'
  id: string
  uid: string
}

import * as ReactApollo from 'react-apollo'

export class GetMusicsWithMapsComponent extends ReactApollo.Query<
  Query,
  GetMusicsWithMapsVariables
> {}
export class GetViewerComponent extends ReactApollo.Query<
  Query,
  GetViewerVariables
> {}
