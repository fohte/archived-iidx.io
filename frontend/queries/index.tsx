export type Maybe<T> = T | null

export enum PlayStyle {
  Sp = 'SP',
  Dp = 'DP',
}

export enum Difficulty {
  Normal = 'NORMAL',
  Hyper = 'HYPER',
  Another = 'ANOTHER',
}

export enum ClearLamp {
  Failed = 'FAILED',
  Assist = 'ASSIST',
  Easy = 'EASY',
  Normal = 'NORMAL',
  Hard = 'HARD',
  ExHard = 'EX_HARD',
  FullCombo = 'FULL_COMBO',
}

export enum Grade {
  A = 'A',
  Aa = 'AA',
  Aaa = 'AAA',
  B = 'B',
  C = 'C',
  D = 'D',
  E = 'E',
  F = 'F',
}

// ====================================================
// Documents
// ====================================================

export type FindMapVariables = {
  id: string
  playStyle: PlayStyle
  difficulty: Difficulty
  username: string
}

export type FindMapQuery = {
  __typename?: 'Query'

  music: Maybe<FindMapMusic>
}

export type FindMapMusic = {
  __typename?: 'Music'

  id: string

  title: string

  subTitle: string

  genre: string

  artist: string

  textageUid: string

  series: number

  leggendaria: boolean

  map: Maybe<FindMapMap>
}

export type FindMapMap = {
  __typename?: 'Map'

  id: string

  numNotes: number

  level: number

  playStyle: PlayStyle

  difficulty: Difficulty

  minBpm: number

  maxBpm: number

  bestResult: Maybe<FindMapBestResult>
}

export type FindMapBestResult = {
  __typename?: 'Result'

  id: string

  score: number

  missCount: number

  clearLamp: ClearLamp

  grade: Grade
}

export type FindUserVariables = {
  screenName: string
}

export type FindUserQuery = {
  __typename?: 'Query'

  user: Maybe<FindUserUser>
}

export type FindUserUser = {
  __typename: 'User'

  name: string
}

export type GetMusicsWithMapsVariables = {}

export type GetMusicsWithMapsQuery = {
  __typename?: 'Query'

  musics: Maybe<GetMusicsWithMapsMusics[]>
}

export type GetMusicsWithMapsMusics = {
  __typename: 'Music'

  id: string

  title: string

  subTitle: string

  maps: GetMusicsWithMapsMaps[]
}

export type GetMusicsWithMapsMaps = {
  __typename: 'Map'

  id: string

  difficulty: Difficulty

  level: number

  playStyle: PlayStyle
}

export type GetViewerVariables = {}

export type GetViewerQuery = {
  __typename?: 'Query'

  viewer: Maybe<GetViewerViewer>
}

export type GetViewerViewer = {
  __typename: 'User'

  id: string

  name: string
}

export type RegisterVariables = {
  username: string
  displayName?: Maybe<string>
}

export type RegisterMutation = {
  __typename?: 'Mutation'

  createUser: Maybe<RegisterCreateUser>
}

export type RegisterCreateUser = {
  __typename?: 'CreateUserPayload'

  user: RegisterUser
}

export type RegisterUser = {
  __typename?: 'User'

  name: string

  profile: RegisterProfile
}

export type RegisterProfile = {
  __typename?: 'UserProfile'

  displayName: string
}

export type RegisterResultsWithCsvVariables = {
  csv: string
  playStyle: PlayStyle
}

export type RegisterResultsWithCsvMutation = {
  __typename?: 'Mutation'

  registerResultsWithCSV: Maybe<RegisterResultsWithCsvRegisterResultsWithCsv>
}

export type RegisterResultsWithCsvRegisterResultsWithCsv = {
  __typename?: 'RegisterResultsWithCSVPayload'

  success: boolean
}

import * as ReactApollo from 'react-apollo'
import * as React from 'react'

import gql from 'graphql-tag'

// ====================================================
// Components
// ====================================================

export const FindMapDocument = gql`
  query findMap(
    $id: ID!
    $playStyle: PlayStyle!
    $difficulty: Difficulty!
    $username: String!
  ) {
    music(id: $id) {
      id
      title
      subTitle
      genre
      artist
      textageUid
      series
      leggendaria
      map(playStyle: $playStyle, difficulty: $difficulty) {
        id
        numNotes
        level
        playStyle
        difficulty
        minBpm
        maxBpm
        bestResult(username: $username) {
          id
          score
          missCount
          clearLamp
          grade
        }
      }
    }
  }
`
export class FindMapComponent extends React.Component<
  Partial<ReactApollo.QueryProps<FindMapQuery, FindMapVariables>>
> {
  render() {
    return (
      <ReactApollo.Query<FindMapQuery, FindMapVariables>
        query={FindMapDocument}
        {...(this as any)['props'] as any}
      />
    )
  }
}
export type FindMapProps<TChildProps = any> = Partial<
  ReactApollo.DataProps<FindMapQuery, FindMapVariables>
> &
  TChildProps
export function FindMapHOC<TProps, TChildProps = any>(
  operationOptions:
    | ReactApollo.OperationOption<
        TProps,
        FindMapQuery,
        FindMapVariables,
        FindMapProps<TChildProps>
      >
    | undefined,
) {
  return ReactApollo.graphql<
    TProps,
    FindMapQuery,
    FindMapVariables,
    FindMapProps<TChildProps>
  >(FindMapDocument, operationOptions)
}
export const FindUserDocument = gql`
  query findUser($screenName: String!) {
    user(name: $screenName) {
      __typename
      name
    }
  }
`
export class FindUserComponent extends React.Component<
  Partial<ReactApollo.QueryProps<FindUserQuery, FindUserVariables>>
> {
  render() {
    return (
      <ReactApollo.Query<FindUserQuery, FindUserVariables>
        query={FindUserDocument}
        {...(this as any)['props'] as any}
      />
    )
  }
}
export type FindUserProps<TChildProps = any> = Partial<
  ReactApollo.DataProps<FindUserQuery, FindUserVariables>
> &
  TChildProps
export function FindUserHOC<TProps, TChildProps = any>(
  operationOptions:
    | ReactApollo.OperationOption<
        TProps,
        FindUserQuery,
        FindUserVariables,
        FindUserProps<TChildProps>
      >
    | undefined,
) {
  return ReactApollo.graphql<
    TProps,
    FindUserQuery,
    FindUserVariables,
    FindUserProps<TChildProps>
  >(FindUserDocument, operationOptions)
}
export const GetMusicsWithMapsDocument = gql`
  query getMusicsWithMaps {
    musics {
      __typename
      id
      title
      subTitle
      maps {
        __typename
        id
        difficulty
        level
        playStyle
      }
    }
  }
`
export class GetMusicsWithMapsComponent extends React.Component<
  Partial<
    ReactApollo.QueryProps<GetMusicsWithMapsQuery, GetMusicsWithMapsVariables>
  >
> {
  render() {
    return (
      <ReactApollo.Query<GetMusicsWithMapsQuery, GetMusicsWithMapsVariables>
        query={GetMusicsWithMapsDocument}
        {...(this as any)['props'] as any}
      />
    )
  }
}
export type GetMusicsWithMapsProps<TChildProps = any> = Partial<
  ReactApollo.DataProps<GetMusicsWithMapsQuery, GetMusicsWithMapsVariables>
> &
  TChildProps
export function GetMusicsWithMapsHOC<TProps, TChildProps = any>(
  operationOptions:
    | ReactApollo.OperationOption<
        TProps,
        GetMusicsWithMapsQuery,
        GetMusicsWithMapsVariables,
        GetMusicsWithMapsProps<TChildProps>
      >
    | undefined,
) {
  return ReactApollo.graphql<
    TProps,
    GetMusicsWithMapsQuery,
    GetMusicsWithMapsVariables,
    GetMusicsWithMapsProps<TChildProps>
  >(GetMusicsWithMapsDocument, operationOptions)
}
export const GetViewerDocument = gql`
  query getViewer {
    viewer {
      __typename
      id
      name
    }
  }
`
export class GetViewerComponent extends React.Component<
  Partial<ReactApollo.QueryProps<GetViewerQuery, GetViewerVariables>>
> {
  render() {
    return (
      <ReactApollo.Query<GetViewerQuery, GetViewerVariables>
        query={GetViewerDocument}
        {...(this as any)['props'] as any}
      />
    )
  }
}
export type GetViewerProps<TChildProps = any> = Partial<
  ReactApollo.DataProps<GetViewerQuery, GetViewerVariables>
> &
  TChildProps
export function GetViewerHOC<TProps, TChildProps = any>(
  operationOptions:
    | ReactApollo.OperationOption<
        TProps,
        GetViewerQuery,
        GetViewerVariables,
        GetViewerProps<TChildProps>
      >
    | undefined,
) {
  return ReactApollo.graphql<
    TProps,
    GetViewerQuery,
    GetViewerVariables,
    GetViewerProps<TChildProps>
  >(GetViewerDocument, operationOptions)
}
export const RegisterDocument = gql`
  mutation register($username: String!, $displayName: String) {
    createUser(username: $username, displayName: $displayName) {
      user {
        name
        profile {
          displayName
        }
      }
    }
  }
`
export class RegisterComponent extends React.Component<
  Partial<ReactApollo.MutationProps<RegisterMutation, RegisterVariables>>
> {
  render() {
    return (
      <ReactApollo.Mutation<RegisterMutation, RegisterVariables>
        mutation={RegisterDocument}
        {...(this as any)['props'] as any}
      />
    )
  }
}
export type RegisterProps<TChildProps = any> = Partial<
  ReactApollo.MutateProps<RegisterMutation, RegisterVariables>
> &
  TChildProps
export type RegisterMutationFn = ReactApollo.MutationFn<
  RegisterMutation,
  RegisterVariables
>
export function RegisterHOC<TProps, TChildProps = any>(
  operationOptions:
    | ReactApollo.OperationOption<
        TProps,
        RegisterMutation,
        RegisterVariables,
        RegisterProps<TChildProps>
      >
    | undefined,
) {
  return ReactApollo.graphql<
    TProps,
    RegisterMutation,
    RegisterVariables,
    RegisterProps<TChildProps>
  >(RegisterDocument, operationOptions)
}
export const RegisterResultsWithCsvDocument = gql`
  mutation registerResultsWithCSV($csv: String!, $playStyle: PlayStyle!) {
    registerResultsWithCSV(csv: $csv, playStyle: $playStyle) {
      success
    }
  }
`
export class RegisterResultsWithCsvComponent extends React.Component<
  Partial<
    ReactApollo.MutationProps<
      RegisterResultsWithCsvMutation,
      RegisterResultsWithCsvVariables
    >
  >
> {
  render() {
    return (
      <ReactApollo.Mutation<
        RegisterResultsWithCsvMutation,
        RegisterResultsWithCsvVariables
      >
        mutation={RegisterResultsWithCsvDocument}
        {...(this as any)['props'] as any}
      />
    )
  }
}
export type RegisterResultsWithCsvProps<TChildProps = any> = Partial<
  ReactApollo.MutateProps<
    RegisterResultsWithCsvMutation,
    RegisterResultsWithCsvVariables
  >
> &
  TChildProps
export type RegisterResultsWithCsvMutationFn = ReactApollo.MutationFn<
  RegisterResultsWithCsvMutation,
  RegisterResultsWithCsvVariables
>
export function RegisterResultsWithCsvHOC<TProps, TChildProps = any>(
  operationOptions:
    | ReactApollo.OperationOption<
        TProps,
        RegisterResultsWithCsvMutation,
        RegisterResultsWithCsvVariables,
        RegisterResultsWithCsvProps<TChildProps>
      >
    | undefined,
) {
  return ReactApollo.graphql<
    TProps,
    RegisterResultsWithCsvMutation,
    RegisterResultsWithCsvVariables,
    RegisterResultsWithCsvProps<TChildProps>
  >(RegisterResultsWithCsvDocument, operationOptions)
}
