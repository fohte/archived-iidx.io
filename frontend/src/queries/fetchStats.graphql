query fetchStats($username: String!, $playStyle: PlayStyle!) {
  user(name: $username) {
    countByEachLevelAndGrade(playStyle: $playStyle) {
      grade
      level
      count
    }
  }
}
