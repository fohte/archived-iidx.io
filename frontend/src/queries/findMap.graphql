query findMap(
  $id: ID!
  $playStyle: PlayStyle!
  $difficulty: Difficulty!
  $username: String!
) {
  music(id: $id) {
    id
    title
    genre
    artist
    textageUid
    series
    leggendaria
    map(playStyle: $playStyle, difficulty: $difficulty) {
      id
      numNotes
      level
      playStyle
      difficulty
      minBpm
      maxBpm
      bestResult(username: $username) {
        id
        score
        missCount
        clearLamp
        grade
      }
    }
  }
}
