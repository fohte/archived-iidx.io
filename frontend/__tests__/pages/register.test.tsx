import * as React from 'react'
import { MockedProvider } from 'react-apollo/test-utils'
import * as renderer from 'react-test-renderer'

import Register from '@app/pages/register'
import { RegisterDocument, RegisterMutation } from '@app/queries'

describe('/register', () => {
  const data: RegisterMutation = {
    createUser: {
      user: {
        name: 'name',
        profile: {
          displayName: 'displayName',
        },
      },
    },
  }

  const mocks = [{ request: { query: RegisterDocument }, result: { data } }]

  it('renders correctly', () => {
    const component = renderer.create(
      <MockedProvider mocks={mocks}>
        <Register />
      </MockedProvider>,
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
