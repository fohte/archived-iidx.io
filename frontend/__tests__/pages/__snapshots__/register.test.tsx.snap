// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/register renders correctly 1`] = `
<div
  className="layout ant-layout"
>
  <div
    className="ant-layout-header"
  >
    <ul
      className="ant-menu ant-menu-dark ant-menu-root ant-menu-vertical"
      role="menu"
      style={
        Object {
          "lineHeight": "64px",
        }
      }
    >
      <div
        className="sc-bwzfXH sUFFa"
        onClick={[Function]}
      >
        <span>
          iidx.io
        </span>
      </div>
      <div
        className="sc-htpNat gEBfLP"
        mode="vertical"
        onClick={[Function]}
        onSelect={[Function]}
      >
        <i
          className="anticon anticon-loading"
        >
          <svg
            aria-hidden="true"
            className="anticon-spin"
            data-icon="loading"
            fill="currentColor"
            height="1em"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 0 0-94.3-139.9 437.71 437.71 0 0 0-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
            />
          </svg>
        </i>
      </div>
    </ul>
  </div>
  <div
    className="ant-layout"
  >
    <div
      className="ant-layout-content"
      style={
        Object {
          "padding": "0 50px",
        }
      }
    >
      <div
        style={
          Object {
            "background": "#fff",
            "minHeight": 280,
            "padding": 24,
          }
        }
      >
        <form
          className="ant-form ant-form-horizontal"
          onSubmit={[Function]}
        >
          <div
            className="ant-row ant-form-item"
          >
            <div
              className="ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control"
              >
                <span
                  className="ant-form-item-children"
                >
                  <span
                    className="ant-input-affix-wrapper"
                  >
                    <span
                      className="ant-input-prefix"
                    >
                      <span
                        style={
                          Object {
                            "color": "rgba(0, 0, 0, 0.25)",
                          }
                        }
                      >
                        @
                      </span>
                    </span>
                    <input
                      className="ant-input"
                      data-__field={
                        Object {
                          "name": "username",
                        }
                      }
                      data-__meta={
                        Object {
                          "name": "username",
                          "originalProps": Object {
                            "disabled": false,
                            "placeholder": "Username",
                            "prefix": <Atmark />,
                            "prefixCls": "ant-input",
                            "type": "text",
                          },
                          "ref": null,
                          "rules": Array [
                            Object {
                              "message": "Please input your username.",
                              "required": true,
                            },
                            Object {
                              "message": "Username may only contain alphanumeric characters or underscores, and cannot begin with a number.",
                              "pattern": /\\^\\[a-zA-Z_\\]\\[a-zA-Z0-9_\\]\\*\\$/,
                            },
                            Object {
                              "max": 20,
                              "message": "Username is too long (maximum is 20 characters).",
                            },
                          ],
                          "trigger": "onChange",
                          "validate": Array [
                            Object {
                              "rules": Array [
                                Object {
                                  "message": "Please input your username.",
                                  "required": true,
                                },
                                Object {
                                  "message": "Username may only contain alphanumeric characters or underscores, and cannot begin with a number.",
                                  "pattern": /\\^\\[a-zA-Z_\\]\\[a-zA-Z0-9_\\]\\*\\$/,
                                },
                                Object {
                                  "max": 20,
                                  "message": "Username is too long (maximum is 20 characters).",
                                },
                              ],
                              "trigger": Array [
                                "onChange",
                              ],
                            },
                          ],
                          "valuePropName": "value",
                        }
                      }
                      disabled={false}
                      id="username"
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Username"
                      style={null}
                      type="text"
                      value=""
                    />
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            className="ant-row ant-form-item"
          >
            <div
              className="ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control"
              >
                <span
                  className="ant-form-item-children"
                >
                  <span
                    className="ant-input-affix-wrapper"
                  >
                    <span
                      className="ant-input-prefix"
                    >
                      <i
                        className="anticon anticon-user sc-bdVaJa eaTnkz"
                      >
                        <svg
                          aria-hidden="true"
                          className=""
                          data-icon="user"
                          fill="currentColor"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                          />
                        </svg>
                      </i>
                    </span>
                    <input
                      className="ant-input"
                      data-__field={
                        Object {
                          "name": "displayName",
                        }
                      }
                      data-__meta={
                        Object {
                          "name": "displayName",
                          "originalProps": Object {
                            "disabled": false,
                            "placeholder": "Display Name",
                            "prefix": <Styled(Icon)
                              type="user"
                            />,
                            "prefixCls": "ant-input",
                            "type": "text",
                          },
                          "ref": null,
                          "rules": Array [
                            Object {
                              "max": 40,
                              "message": "Display Name is too long (maximum is 40 characters).",
                            },
                          ],
                          "trigger": "onChange",
                          "validate": Array [
                            Object {
                              "rules": Array [
                                Object {
                                  "max": 40,
                                  "message": "Display Name is too long (maximum is 40 characters).",
                                },
                              ],
                              "trigger": Array [
                                "onChange",
                              ],
                            },
                          ],
                          "valuePropName": "value",
                        }
                      }
                      disabled={false}
                      id="displayName"
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Display Name"
                      style={null}
                      type="text"
                      value=""
                    />
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            className="ant-row ant-form-item"
          >
            <div
              className="ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control"
              >
                <span
                  className="ant-form-item-children"
                >
                  <button
                    className="ant-btn ant-btn-primary"
                    onClick={[Function]}
                    type="submit"
                  >
                    <span>
                      Sign up
                    </span>
                  </button>
                </span>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div
      className="ant-layout-footer"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <i
        className="anticon anticon-copyright"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="copyright"
          fill="currentColor"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372zm5.6-532.7c53 0 89 33.8 93 83.4.3 4.2 3.8 7.4 8 7.4h56.7c2.6 0 4.7-2.1 4.7-4.7 0-86.7-68.4-147.4-162.7-147.4C407.4 290 344 364.2 344 486.8v52.3C344 660.8 407.4 734 517.3 734c94 0 162.7-58.8 162.7-141.4 0-2.6-2.1-4.7-4.7-4.7h-56.8c-4.2 0-7.6 3.2-8 7.3-4.2 46.1-40.1 77.8-93 77.8-65.3 0-102.1-47.9-102.1-133.6v-52.6c.1-87 37-135.5 102.2-135.5z"
          />
        </svg>
      </i>
       
      2018
       iidx.io
    </div>
  </div>
</div>
`;
